package DataCommon;

//装备商店内的道具
message EquipShopCPP
{
	required uint32 eqid			= 1;	//装备id
	required bool status			= 2;	//状态
	repeated uint32 subs			= 3;	//洗练属性
}

//返回已购买的装备信息
message EquipSubsCPP
{
	required uint32 type		= 1;	//洗练属性类型
	required uint32 level		= 2;  //洗练属性等级
}

message EquipmentCPP
{
	required uint32 	equd 				= 1;	//物品ud
	required uint32 	eqid				= 2;	//物品id
	required uint32 	num					= 3;	//物品数目
	required uint32 	deltime				= 4;	//物品删除时间
	required uint32 	heroid				= 5;	//英雄id
	required uint32 	star				= 7;	//物品星级
	required bool	 	issold				= 8;	//物品是否出售标志
	repeated uint32 	bornval				= 9;	//物品的统和勇属性值，没有则为空
	repeated EquipSubsCPP 	subs				= 10;   //物品的洗练属性，没有则为空
}

message EquipmentAllCPP
{
	repeated EquipmentCPP  equips	= 1;  //批量物品信息
}

message RewardItemCPP
{
	required	uint32 	value		= 1;  //奖励值
	required	uint64	total_value	= 2;  //对应的最新值 
}

message CommonBaseRewardCPP
{
	optional RewardItemCPP food			= 1;	//食物
	optional RewardItemCPP iron			= 2;	//镔铁
	optional RewardItemCPP order		= 3;  //募兵令
	optional RewardItemCPP exp			= 4;  //如果用户升级了会额外推送升级协议
	optional RewardItemCPP coin			= 5; //钞票
	optional RewardItemCPP wood			= 6; //钢材
	optional RewardItemCPP hammer		= 7; //黄金锤子	
	optional RewardItemCPP cash			= 8; //钻石
	optional RewardItemCPP refresh		= 9; //免费洗练次数
	optional RewardItemCPP arefresh		= 10; //至尊洗练次数
	optional RewardItemCPP sacrifice	= 11; //祭祀次数
	optional RewardItemCPP market		= 12; //集市次数
	optional RewardItemCPP shadow		= 13; //影子次数
}

message UserResourceCPP
{
	required uint32	 coin		= 1;	//铜币
	required uint32	 wood		= 2;	//木材
	required uint32	 food		= 3;	//粮食
	required uint32	 iron		= 4;	//镔铁
	optional uint32	 cash		= 5; 	//钻石
}


message CostItemCPP
{
	required uint32 value		= 1; //扣除值
	required uint64	total_value	= 2;  //当前值
}
